handler/handler.ts

```diff
@@ -1,28 +1,47 @@
+//handler/handler.ts
+
 import { Request, Response } from 'express';
-import { getUsers } from '../controller/controller';
-import jwt from 'jsonwebtoken'; // Importar el módulo jsonwebtoken
+import jwt from 'jsonwebtoken';
+import { UsersController } from '../controller/controller';
+import { TransformedUser, TransformedPosts } from '../models/models';
 
 const SECRET_KEY = 'gauss626';
 
-const getUsersHandler = async (req: Request, res: Response) => {
-  try {
-    const transformedUsers = await getUsers();
-    res.json(transformedUsers);
-  } catch (error) {
-    console.error('Error al llamar a la API:', error);
-    res.sendStatus(500); // Internal Server Error
+export class Handler {
+  private usersController: UsersController;
+
+  constructor() {
+    this.usersController = new UsersController();
   }
-};
 
-const loginHandler = (req: Request, res: Response) => {
-  // Generar y devolver el token JWT en la respuesta
-  const payload = {
-    username: 'example',
-    role: 'admin'
-  };
+  async getUsersHandler(req: Request, res: Response) {
+    try {
```

Comment:
This check shouldn't be here in the handler, it should be as part of the middleware.